lib/zendesk_api/actions.rb

```diff
@@ -27,14 +27,14 @@ def save!(options = {})
 
       req_path = options[:path] if options[:path]
 
-      save_associations
-
       @response = @client.connection.send(method, req_path) do |req|
         req.body = attributes_for_save.merge(@global_params)
 
         yield req if block_given?
       end
 
+      save_associations
+
```

Comment:
ATTENTION! This is the change for this PR:

From the issue 420 (not linking it until I have a proper solution):

> However, note that the tags on the ticket are only "tag_via_gem" - the tag change was persisted despite safe_update being set to true.

This happens because we were saving the associations before the resource, and we don't have transactions on APIs.

Reply:
Rather than a global option, I was considering a "per request" option, like the above `ticket.save(save_associations: :after)`. That would allow users to fix their problem on the specific call, and not having to review the whole code base.

Solving the second problem using a PUT opens the new one: using PUT instead of POST results in a union of the tags. This would change the behaviour for all our clients. Adding `safe_update` on POST seems like another workaround to not having transactional APIs. Other options are discussed in the card RED-1545.

I'd like to solve the specific problem in the Z2 PT, and leave this second point that no-one yet complained about for another day.